dags/synapse-datasets-created-7-days.py

```diff
@@ -0,0 +1,173 @@
+It is scheduled to run every Monday at 00:00 UTC.
+"""
+
+from dataclasses import dataclass
+from datetime import date, datetime
+from typing import List
+
+import synapseclient
+from airflow.decorators import dag, task
+from airflow.models import Variable
+from airflow.models.param import Param
+from airflow.providers.snowflake.hooks.snowflake import SnowflakeHook
+from orca.services.synapse import SynapseHook
+from slack_sdk import WebClient
+
+dag_params = {
+    "snowflake_conn_id": Param("SNOWFLAKE_SYSADMIN_PORTAL_RAW_CONN", type="string"),
+    "synapse_conn_id": Param("SYNAPSE_ORCA_SERVICE_ACCOUNT_CONN", type="string"),
+    "current_date": Param(date.today().strftime("%Y-%m-%d"), type="string"),
+}
+
+dag_config = {
+    "schedule_interval": "0 0 * * 1",
+    "start_date": datetime(2025, 2, 1),
+    "catchup": False,
+    "default_args": {
+        "retries": 1,
+    },
+    "tags": ["snowflake"],
+    "params": dag_params,
+}
+
+SYNAPSE_RESULTS_TABLE = "syn64919238"
+
+
+@dataclass
+class EntityCreated:
+    """Dataclass to record
+
+    Attributes:
+        name: The name of this entity.
+        id: The unique immutable ID for this entity.
+        parent_id: parent id of the entity
+        node_type: Type of Node
+        content_type: Content type of folder annotated,
+        created_on: The date this entity was created.
+        created_by: The ID of the user that created this entity.
+        is_public: True to indicate if this entity is public.
+    """
+
+    name: str
+    id: int
+    parent_id: int
+    node_type: str
+    content_type: str
+    created_on: str
+    created_by: str
+    is_public: bool
+
+
+@dag(**dag_config)
+def datasets_or_projects_created_7_days() -> None:
+    """Execute a query on Snowflake and report the results to a Synapse table."""
+
+    @task
+    def get_datasets_projects_created_7_days(**context) -> List[EntityCreated]:
+        """Execute the query on Snowflake and return the results."""
+        snow_hook = SnowflakeHook(context["params"]["snowflake_conn_id"])
+        ctx = snow_hook.get_conn()
+        cs = ctx.cursor()
+        query = f"""
+        SELECT 
+            name,
+            id,
+            parent_id,
+            node_type,
+            annotations:annotations:contentType:value as content_type,
+            TO_DATE(created_on) as entity_created_date,
+            created_by,
+            is_public,
+        FROM 
+            synapse_data_warehouse.synapse.node_latest
+        WHERE 
+            (
+                node_type IN ('dataset', 'project') 
+                OR 
+                (content_type LIKE '%dataset%' AND content_type IS NOT NULL)
+            )
+        AND 
+            (
+                entity_created_date >= CURRENT_TIMESTAMP - INTERVAL '7 days' 
+            )
+        ORDER BY entity_created_date;
+        """
+        print(query)
```

Comment:
Was this print statement for troubleshooting?

Reply:
IMO this is probably fine. The consequence is that it'll be a message that gets printed to the Airflow logs, but there isn't anything sensitive here. It can be the first part of investigating an issue if something pops up (making sure the query is correct).

Reply:
I saw some other files, such as synapse-by-the-numbers-dag have the print statement so I kept it here. I think this might help with trouble shooting if we are going to insert backfill date in the query

Reply:
Fine with me!